File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min = prices[0]
        res = 0
        for i in range(len(prices)):
            num = prices[i]
            if min > num:
                min = num
            curr = num - min
            if res < curr:
                res = curr
                
        return res

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = prices[0]
    let max = 0
    
    for (let p of prices) {
        max = Math.max(max, p - min)
        min = Math.min(min, p)
    }
    return max
};